src/lib.rs

```diff
@@ -124,6 +125,44 @@ pub fn get_worker_id() -> usize {
 /// Socket used for coordinating a Gaggle distributed load test.
 pub(crate) struct Socket {}
 
+/// Wrapper over File object that contains context for errors
+pub(crate) struct ReportFile {
+    file: Option<File>,
+    path: String,
+    command_line_option: String,
+}
+
+impl ReportFile {
+    async fn create(&mut self) -> Result<(), GooseError> {
+        let file = File::create(&self.path)
+            .await
+            .map_err(|e| GooseError::InvalidOption {
+                option: self.command_line_option.clone(),
+                value: self.path.clone(),
+                detail: format!("Failed to create report file: {}", e),
+            })?;
+        self.file = Some(file);
+        Ok(())
+    }
+
+    pub(crate) async fn write_all_and_flush(&mut self, report: &[u8]) -> Result<(), GooseError> {
+        let file = self.file.as_mut().unwrap();
```

Comment:
I'm leery of having an `.unwrap()` here. At minimum, please add a comment explaining why this is safe.

Reply:
Removed the filed, the files is now created here